tsdb/head_test.go

```diff
@@ -3019,7 +3018,12 @@ func TestHistogramInWALAndMmapChunk(t *testing.T) {
 			h.NegativeBuckets = h.PositiveBuckets
 			_, err := app.AppendHistogram(0, s2, int64(ts), h, nil)
 			require.NoError(t, err)
-			exp[k2] = append(exp[k2], sample{t: int64(ts), h: h.Copy()})
+			eh := h.Copy()
+			if !gauge && ts%30 == 1 {
```

Comment:
Did you mean this? (If yes, should the tests be failing?)
```suggestion
			if !gauge && ts%20 == 1 {
```

Reply:
The code has a problem here, indeed, but it's different from what you think. Your comment made me think it through again, and this time more carefully, with the following result:

When we reach `ts==20`, we add 10 float samples in the inner loop, which also increments `ts`. After that, we'll go into the next iteration of the outer loop. It will add the first histogram after a float sample. It does this with `ts==31`. So here, the test code is doing the right thing.

Once `ts` hits `40`, another 10 float samples are added, and `ts` is `50` after that, or `51` in the next iteration of the outer loop. The test code does _not_ catch that and therefore does not require an "unknown" reset hint at this time. However, because new chunks being cut would create the same "unknown" reset hint, the test tolerates additional "unknown" reset hints. So this one isn't making the test fail.

Going on, `ts` hits `60` eventually, again triggering the addition of 10 float samples. The next histogram after it is added at `71`, with the same outcome as above, i.e. the "unknown" reset hint is simply tolerated, but not positively verified.

This goes on two more times until we hit `ts==91`, at which time we again _require_ an "unknown" counter reset, and here everything works again, more or less by coincidence.

To improve the verification, we need to adjust the condition to `ts > 30 && (ts-10)%20 == 1`, which I have done now.